ldt/params/AquaCrop/define_AC_compartments.F90

```diff
@@ -0,0 +1,212 @@
+  use AquaCrop_parmsMod
+  use ac_utils
+
+  implicit none
+! !ARGUMENTS: 
+  integer, intent(in) :: n
+  real, intent(inout) :: array(LDT_rc%lnc(n),LDT_rc%lnr(n),12)
+
+! !LOCALS:
+  integer           :: i,j,k,num_types,ct,rc,nrcomp
+  integer           :: ftn, ftn1, r,c, ios1
+  real, allocatable :: depths(:)
+  real, allocatable :: rootingd(:)
+  real, allocatable :: csizes(:,:)
+  real              :: TotalDepthL, TotDepthC, DeltaZ, fAdd
+  character(200)    :: dir
+  character(200)    :: cropinv_file, crop_file
+  character(100)    :: header1, str1
+  character(100)    :: read_cropname
+  character(100)    :: read_fullname
+
+
+! !DESCRIPTION:
+!  This subroutine sets a constant crop type defined in the
+!  configuration file.
+!
+!  The arguments are:
+!  \begin{description}
+!  \item[n]
+!   index of the nest
+!  \item[array]
+!   output field with the compartment size for the 12 levels
+!  \end{description}
+!EOP
+! ____________________________
+
+    !- Get number of layers
+    call ESMF_ConfigFindLabel(LDT_config,"AquaCrop number of soil layers:",rc=rc)
+    call ESMF_ConfigGetAttribute(LDT_config,Aquacrop_struc(n)%nlayers,rc=rc)
+    call LDT_verify(rc,"AquaCrop number of soil layers: not defined")
+    !- Get thickness
+    allocate(depths(Aquacrop_struc(n)%nlayers))
+    call ESMF_ConfigFindLabel(LDT_config,"AquaCrop soil layer thickness:",rc=rc)
+    call ESMF_ConfigGetAttribute(LDT_config,depths,rc=rc)
+    call LDT_verify(rc,"AquaCrop soil layer thickness: not defined")
+
+    do i = 1, AquaCrop_struc(n)%nlayers
+        AquaCrop_struc(n)%lthickness(i) = depths(i)
+    enddo
+
+    write(LDT_logunit,*) "[INFO] AquaCrop number of soil layers: ", Aquacrop_struc(n)%nlayers
+    do i=1,Aquacrop_struc(n)%nlayers
+        write (str1, *) i
+        write(LDT_logunit,*) "Depth layer "//trim(str1)//" =", AquaCrop_struc(n)%lthickness(i)
+    enddo
+
+    !- Get max rooting depth
+    !- Finds crop inventory
+    call ESMF_ConfigFindLabel(LDT_config,"AquaCrop crop library directory:",rc=rc)
+    call ESMF_ConfigGetAttribute(LDT_Config, dir,rc=rc)
+    call LDT_verify(rc,"AquaCrop crop library directory: not defined")
+
+    cropinv_file = trim(dir)//"AC_Crop.Inventory"
+    !- Open file:
+    ftn = LDT_getNextUnitNumber()
+    open(ftn, file=cropinv_file, status='old', form='formatted',&
+            iostat=ios1)
+    call LDT_verify(ios1,"AquaCrop crop inventory does not exist")
+    !- Read crop inventory file:
+    read(ftn,fmt=*) num_types
+    read(ftn,fmt=*) header1
+    allocate(rootingd(num_types))
+    do i = 1, num_types
+        read(ftn,fmt=*) k, read_cropname, read_fullname 
+        ! Open corresponding crop file
+        crop_file = trim(dir)//"AC_Crop.Files/"//trim(read_cropname)//".CRO"
+        ftn1 = LDT_getNextUnitNumber()
+        open(ftn1, file=crop_file, status='old', form='formatted',&
+                iostat=ios1)
+        call LDT_verify(ios1,"AquaCrop crop file does not exist")
+        do j=1,37
```

Comment:
Yes, I know. I think this is what we agreed on. Dirk usually does not change the order of the crop file, some lines become obsolete but are still read. 
Anyway, for a new version in LIS, this will need to be checked.